File: TresEnRaya/TresEnRayaServidor/Entities/Jugador.cs
namespace TresEnRayaServidor.Entities
{
    public class Jugador
    {
        #region Atributos
        private string connectionId;
        private string simbolo;
        private string nombre;
        #endregion

        #region Constructores
        public Jugador() { }

        public Jugador(string connectionId, string simbolo, string nombre)
        {
            this.connectionId = connectionId;
            this.simbolo = simbolo;
            this.nombre = nombre;
        }
        #endregion

        #region Propiedades
        public string ConnectionId
        {
            get { return connectionId; }
            set { connectionId = value; }
        }

        public string Simbolo
        {
            get { return simbolo; }
            set { simbolo = value; }
        }

        public string Nombre
        {
            get { return nombre; }
            set { nombre = value; }
        }
        #endregion

        #region Métodos
        public bool EsValido()
        {
            return !string.IsNullOrEmpty(connectionId) && !string.IsNullOrEmpty(simbolo);
        }
        #endregion
    }
}

File: TresEnRaya/TresEnRayaAplicacion/src/Domain/Entities/Jugador.ts
export class Jugador {
  private _connectionId: string;
  private _simbolo: string;
  private _nombre: string;

  constructor(connectionId: string, simbolo: string, nombre: string) {
    this._connectionId = connectionId;
    this._simbolo = simbolo;
    this._nombre = nombre;
  }

  public get connectionId(): string {
    return this._connectionId;
  }

  public set connectionId(value: string) {
    this._connectionId = value;
  }

  public get simbolo(): string {
    return this._simbolo;
  }

  public set simbolo(value: string) {
    this._simbolo = value;
  }

  public get nombre(): string {
    return this._nombre;
  }

  public set nombre(value: string) {
    this._nombre = value;
  }

  public esValido(): boolean {
    return (
      this.connectionId.length > 0 &&
      (this.simbolo === "X" || this.simbolo === "O")
    );
  }

  public static fromJSON(json: any): Jugador {
    return new Jugador(
      json.connectionId || "",
      json.simbolo || "",
      json.nombre || "Jugador"
    );
  }

  public toJSON(): any {
    return {
      connectionId: this.connectionId,
      simbolo: this.simbolo,
      nombre: this.nombre,
    };
  }
}